compiler/src/iree/compiler/Dialect/Stream/Transforms/test/specialize_encodings.mlir

```diff
@@ -234,6 +234,32 @@ util.global private @device_a = #device_target_local_0_
 
 // -----
 
+// Test that the encoding is dropped if the pass can not interpret the encoding.
+// E.g., the `encoding` attribute is not present in the target configuration, or
+// the `encoding` attribute does not implement the EncodingLayoutAttrInterface
+// interface.
+
+#map0 = affine_map<(m, n, k) -> (m, k)>
+#map1 = affine_map<(m, n, k) -> (k, n)>
+#map2 = affine_map<(m, n, k) -> (m, n)>
+#executable_target_vmvx_bytecode_fb = #hal.executable.target<"vmvx", "vmvx-bytecode-fb", {}>
+#device_target_local_0_ = #hal.device.target<"local", {ordinal = 0 : index}, [#executable_target_vmvx_bytecode_fb]> : !hal.device
+#encoding = #iree_encoding.encoding<operand_index = 0 : index, op_type =  matmul, element_types = [f32, f32, f32], user_indexing_maps = [#map0, #map1, #map2]>
```

Comment:
Similar to how you have the specialized/unspecialized encoding attrs, an unknown encoding attr would be useful for tests like this (that always returned unsupported/etc) - relying on HAL devices and such in here makes the tests more complicated and brittle (as it's relying on the behavior of a different part of the codebase unrelated to what you're doing here)

Reply:
Okay, let me prepare the PR and I can rebase later.

Reply:
https://github.com/iree-org/iree/pull/19986 is out for review.